From 4152916ca78f3fddf418ecc833f0c91050bbb4a2 Mon Sep 17 00:00:00 2001
From: Sean Cross <sean@xobs.io>
Date: Wed, 17 May 2017 10:39:15 +0800
Subject: [PATCH] autoconf: use non-deprecated functions for lssl, lcrypto test

With OpenSSL 1.1, many deprecated functions were removed.  Unfortunately,
the current configure script checks for these functions, causing an error
during compiling.

Change the script to look for non-deprecated functions, which exist
on both 1.0 and 1.1.

This fixes the build on Debian and Arch.

Signed-off-by: Sean Cross <sean@xobs.io>
---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index dac0694..6e0d5a5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -49,8 +49,8 @@ AC_ARG_WITH([openssl],AC_HELP_STRING([--with-openssl], [build with OpenSSL suppo
    [openssl="$withval"],[openssl="no"])
 
 if test "$openssl" = 'yes'; then
-  AC_CHECK_LIB([ssl], [SSL_library_init],,[AC_MSG_ERROR([ssl library missing])])
-  AC_CHECK_LIB([crypto], [CRYPTO_num_locks],,[AC_MSG_ERROR([crypto library missing])])
+  AC_CHECK_LIB([ssl], [SSL_CTX_new],,[AC_MSG_ERROR([ssl library missing])])
+  AC_CHECK_LIB([crypto], [CRYPTO_free],,[AC_MSG_ERROR([crypto library missing])])
 fi
 
 # GeoIP
-- 
2.19.1

